### PR TITLE
Update rails to 6.1

### LINKED ISSUE #115: Upgrade to Rails 6.1
Rails 6.0 is very close to end of live and only receives critical security fixes. See https://guides.rubyonrails.org/maintenance_policy.html

### PR DESCRIPTION
**Pull request by rnestler**
Fixes #115 

The configuration changes where done with `bin/rails app:update` and then manually edited.

### CODE REVIEW

File: config/initializers/cookies_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Be sure to restart your server when you modify this file.
+
+# Specify a serializer for the signed and encrypted cookie jars.
+# Valid options are :json, :marshal, and :hybrid.
+Rails.application.config.action_dispatch.cookies_serializer = :marshal
```
Comment (schmijos) [inline config/initializers/cookies_serializer.rb:5]: Using `:json` should be safe in this app.

File: config/application.rb [hunk 2]
```diff
@@ -20,15 +22,12 @@
 module CoredumpWaterSensorApi
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults "6.0"
+    config.load_defaults 6.0
```
Comment (schmijos) [inline config/application.rb:25]: Did you try loading 6.1 defaults directly? This app is not very complex, so it shouldn't be a problem. If you do so, you can also delete the generated config initializer.

**Review by schmijos — APPROVED** [review]
Since the only web part is the admin interface, I think you could accelerate the migration of the new framework defaults. The api part should not be affected by most of the changes.

File: config/initializers/cookies_serializer.rb (hunk 1 above)
Comment (rnestler) [reply config/initializers/cookies_serializer.rb]: Switched to `:hybrid`. We can still change later.

File: config/application.rb (hunk 2 above)
Comment (rnestler) [reply config/application.rb:25]: I migrated most of the options, but left out the ones which aren't backwards compatible.

File: config/initializers/cookies_serializer.rb (hunk 1 above)
Comment (schmijos) [reply config/initializers/cookies_serializer.rb]: Who logs in? Only @dbrgn does. So we should be fine with `json`.

File: config/application.rb (hunk 2 above)
Comment (schmijos) [reply config/application.rb:25]: This app consists only of the api and the admin interface. I'd carelessly opt-in for the defaults until @dbrgn complains (maybe give him a heads up).

Comment (rnestler) [reply config/application.rb:25]: Well the admin interface currently doesn't work with the 6.1 upgrade due to a assets pipeline issue in production. So I guess I have to rollback...

Comment (rnestler) [reply config/application.rb:25]: I made a hotfix: #119

**Pull request merged**